### PR TITLE
[Feature] Add `databricks_query` resource instead of `databricks_sql_query`

### PR DESCRIPTION
**Pull request by alexott**
## Changes

This PR is built on top of #4051 which should be merged first.

The new resource uses the new [Queries API](https://docs.databricks.com/api/workspace/queries/create) instead of the legacy one that will be deprecated. Since the new resource has a slightly different set of parameters, it was decided to create a new resource and deprecate the old one.

This resource uses old TF SDK to be compatible with TF exporter (until #4050 is implemented).

TODOs:

- Need to discuss how to handle permissions - `sql_query` permissions look like working, but not sure if we should continue to use that API
- Support in the exporter will be in a separate PR

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: docs/resources/query.md
Comment (mgyucht) [inline docs/resources/query.md:39]: Just curious: in other docs, these blocks are often documented as their own headings. Is there any need for consistency around these aspects?

File: docs/resources/query.md
Comment (mgyucht) [inline docs/resources/query.md:132]: Ditto on these w.r.t. using import/removed blocks

File: docs/resources/query.md
Comment (mgyucht) [inline docs/resources/query.md:140]: Great that we can use this as-is again.

File: internal/acceptance/query_test.go
Comment (mgyucht) [inline internal/acceptance/query_test.go:16]: Ditto, let's move this to permissions_test.go.

File: sql/resource_query.go
Comment (mgyucht) [inline sql/resource_query.go:115]: is this another set of APIs where the owner cannot be set on create?

**Review by mgyucht — COMMENTED** [review]
Overall seems good to me. I'll wait to approve until databricks_alert is merged.

File: sql/resource_query.go
Comment (alexott) [reply sql/resource_query.go:115]: yep - same as for alerts

File: docs/resources/query.md
Comment (alexott) [reply docs/resources/query.md:39]: that's a good question - sometimes, moving to a separate block will add an unnecessary context switch. But agree, we need to unify this

File: sql/resource_query.go
Comment (mgyucht) [inline sql/resource_query.go:152]: For Terraform, because DataToStructPointer has values for all fields, can we just use `*` for updateMask?

**Review by mgyucht — APPROVED** [review]
Overall LGTM, just a couple of smaller questions.

File: sql/resource_alert.go
Comment (alexott) [reply sql/resource_alert.go:22]: Because it's covered by previous line

File: sql/resource_alert.go
Comment (alexott) [reply sql/resource_alert.go:41]: We get default from the API, and if we use `suppress_diff` here, then we can't set it to the `false`

File: sql/resource_query.go
Comment (alexott) [reply sql/resource_query.go:152]: API spec says:
> To specify multiple fields in the field mask, use a comma as the separator (no space).

I'm not sure that `*` is supported.

Especially for `owner_user_name` if we don't have change, it will decrease a pressure on backend, as changing the owner is heavy operation (at least as I understood from the explanation of the corresponding team).

**Comment by rauchy** [comment]
Merging for now to unblock releases.

**Pull request merged**
